### ISSUE TITLE
`jest/require-hook` false positives

### ISSUE THREAD

**Issue by kripod-wise** [transferwise/eslint-config#58]
The [`jest/require-hook`](https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/require-hook.md) linting rule is always enforced, despite not being used in test files, but just plain old `.js`, e.g.:

```js
import { TwIconsModule } from '@transferwise/icons/lib/angular';
import angular from 'angular';
import { angular2react } from 'angular2react';

const jQuery = require('jquery/dist/jquery.min');
window._ = require('underscore');

window.$ = jQuery;
window.jQuery = jQuery;

/* Linting errors are thrown when using `require` below */
require('angular-translate');
require('angular-cookies');
require('angular-sanitize');
require('angular-translate-interpolation-messageformat');
require('angular-ui-bootstrap');
require('ng-file-upload');
require('bootstrap/js/dropdown');
require('@transferwise/frontend-common/dist/tw-common');
require('@transferwise/styleguide-components/js/styleguide-components');
require('@transferwise/styleguide-components/js/json-schema');
require('@transferwise/verification-components/verification-components');

// …
```

I think this could be mitigated by only applying the `jest/require-hook` rule on test files as defined by the [default `testMatch` of Jest](https://jestjs.io/docs/configuration#testmatch-arraystring): `["**/__tests__/**/*.[jt]s?(x)", "**/?(*.)+(spec|test).[jt]s?(x)"]`

**Comment by jacklesliewise** [comment]
Thanks for creating the issue! Fixed this in https://github.com/transferwise/eslint-config/pull/62
